### DISCUSSION TITLE
0.8.0 Eaton 5E not found

### DISCUSSION THREAD

**Discussion by dixonbg** [hassio-addons/app-nut#176] [Q&A] [answered]
# Problem/Motivation

I have Eaton 5E USB connected to Raspberry Pi 3 with latest HassIO. 
The addon fails to start, as if the driver does not have permission to the device.
I see that all similar issues were closed in previous versions.  
But it is not resolved for me. And I have no idea how to fix it. Please, help.

## Expected behavior

Addon should start successfully.

## Actual behavior

Addon fails with following log entries
```
[00:18:28] INFO: Connected USB devices:
Bus 001 Device 004: ID 0463:ffff MGE UPS Systems UPS
Bus 001 Device 003: ID 0424:ec00 Standard Microsystems Corp. SMSC9512/9514 Fast Ethernet Adapter
Bus 001 Device 002: ID 0424:9514 Standard Microsystems Corp. SMC9514 Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
[00:18:28] INFO: Generating /etc/nut/upsd.users...
[00:18:28] INFO: Configuring user: admin
[00:18:28] INFO: Password is NOT in the Have I Been Pwned database! Nice!
[00:18:29] INFO: Configuring Device named myups...
[00:18:29] INFO: Starting the UPS drivers...
No matching HID UPS found
Network UPS Tools - Generic HID driver 0.41 (2.7.4)
USB communication driver 0.33
Driver failed to start (exit status=1)
Network UPS Tools - UPS driver controller 2.7.4
[cont-init.d] nut.sh: exited 1.
```

## Steps to reproduce

I used this configuration:
```
users:
  - username: admin
    password: Very_Strong_Pass!
    instcmds:
      - all
    actions:
      - set
      - fsd
devices:
  - name: myups
    driver: usbhid-ups
    port: auto
    config:
      - vendor = '0463*'
mode: netserver
shutdown_host: 'false'
list_usb_devices: 'true'
```
```
BUSNUM: '001'
DEVNAME: /dev/bus/usb/001/004
DEVNUM: '004'
DEVPATH: /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.5
DEVTYPE: usb_device
DRIVER: usb
ID_BUS: usb
ID_FOR_SEAT: usb-platform-3f980000_usb-usb-0_1_5
ID_MODEL: 5E
ID_MODEL_ENC: 5E
ID_MODEL_ID: ffff
ID_PATH: platform-3f980000.usb-usb-0:1.5
ID_PATH_TAG: platform-3f980000_usb-usb-0_1_5
ID_REVISION: '0001'
ID_SERIAL: EATON_5E
ID_USB_INTERFACES: ':030000:'
ID_VENDOR: EATON
ID_VENDOR_ENC: EATON
ID_VENDOR_ID: '0463'
MAJOR: '189'
MINOR: '3'
PRODUCT: 463/ffff/1
SUBSYSTEM: usb
TAGS: ':seat:'
TYPE: 0/0/0
USEC_INITIALIZED: '6495259'
```

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

**Comment by sinclairpaul** [comment]
> - vendor = '0463*'

If you want to use this, it should be `vendorid = 0463*` as per the docs https://github.com/hassio-addons/addon-nut/blob/main/nut/DOCS.md#sub-option-config

The Vendor option would be the name, although if you have a single UPS connected I would remove both.

Assuming you have tried this previously, I would recommend you validate the USB cable and maybe take a look at the comments in #15

**Comment by dixonbg** [comment]
Thank you for your quick reply.
Yes, I have tried so many configurations before, that it seems my last vendor filter I've tried is wrong. Most tests were performed without vendor settings.
I cannot change the cable now, because the system is on a remote location. It could be possible in a month or so.
But I don't see similar errors in my dmesg as reported in the linked issue #15. Also, my raspberry pi is version 3 and the reported bug in VLI eeprom is related to Pi 4.
I'll review post by post again tomorrow and try every possible solution, but at first look I don't see it as related.

**Comment by sinclairpaul** [comment]
There isn't likely anything we can do from an addon perspective, its will be hardware/config issues.  Converting this to a discussion.

**Comment by dixonbg** [comment, marked as answer]
I have managed to fix my issue... it was lack of permissions as I have suggested. But how I have fixed it is way more interesting. Just restart!

You have got to put in giant big red letters right in the beginning of the documentation "Restart Supervisor after plugging in USB devices!" 
It seem supervisor sets the proper permissions to USB devices during boot before exposing them to Add-ons.

**Comment by sinclairpaul** [comment]
This would be something for the Supervisor/HA documentation for device handling, nothing can be done from an addon perspective.
